src/revocation.rs

```diff
@@ -0,0 +1,134 @@
+#[macro_use]
+use log::*;
+use pretty_env_logger;
+
+use crate::common::config_get;
+use crate::crypto;
+use crate::error;
+use crate::error::{Error, Result};
+use crate::secure_mount;
+
+use std::path::Path;
+
+use serde_json::Value;
+
+/// Handles revocation messages via 0mq
+/// See:
+/// - URL: https://github.com/keylime/keylime/blob/master/keylime/revocation_notifier.py
+///   Function: await_notifications
+pub(crate) async fn run_revocation_service() -> Result<()> {
+    let mount = secure_mount::mount()?;
+    let revocation_cert_path =
+        format!("{}/unzipped/RevocationNotifier-cert.crt", mount);
+
+    let mut cert_key;
+
+    // Connect to the service via 0mq
+    let context = zmq::Context::new();
+    let mysock = context
+        .socket(zmq::SUB)
+        .expect("unable to create 0mq socket");
+
+    mysock
+        .set_subscribe(b"")
+        .expect("failed setting the 0mq subscription");
+
+    let revocation_ip =
+        config_get("/etc/keylime.conf", "general", "receive_revocation_ip")?;
+    let revocation_port = config_get(
+        "/etc/keylime.conf",
+        "general",
+        "receive_revocation_port",
+    )?;
+    let endpoint = format!("{}:{}", revocation_ip, revocation_port);
+
+    mysock
+        .connect(endpoint.as_str())
+        .expect("could not connect to revocation service");
+
+    // Unlike the python agent we do not attempt lazy loading. We either
+    // have the certificate, or we don't. If we don't have a key or can't load
+    // the key we return a Configuration error as the service will not work.
+    if Path::new(&revocation_cert_path).exists() {
```

Comment:
```suggestion
    let cert_key = if Path::new(&revocation_cert_path).exists() {
```

Reply:
If you take this, together with the following two patches, you don't need the `let mut cert_key`, as you can set it in the single block of code! Rust is awesome, because an `if ... { V }` statement returns the `V` value!